tests/test_api.py

```diff
@@ -389,3 +389,198 @@ def test_missing_config(config_override):
+    assert actual_ecs_yr == valid_tcr_ecs_diagnosis_results["ecs_yr"]
+
+    test_time = test_CO2_data.index.values
+    for year_to_break in [
+        test_time[0],
+        test_time[15],
+        test_time[115],
+        test_time[-1] - 100,
+        test_time[-1],
+    ]:
+        broken_CO2_data = test_CO2_data.copy()
+        broken_CO2_data.loc[year_to_break] = test_CO2_data.loc[year_to_break] * 1.01
+        with pytest.raises(ValueError, match=r"The TCR/ECS CO2 concs look wrong.*"):
+            magicc_base._get_tcr_ecs_yr_from_CO2_concs(broken_CO2_data)
+
+
+def test_check_tcr_ecs_total_RF(valid_tcr_ecs_diagnosis_results, magicc_base):
+    test_RF_data = valid_tcr_ecs_diagnosis_results["mock_results"][
+        "TOTAL_INCLVOLCANIC_RF"
+    ]["GLOBAL"]
+    magicc_base._check_tcr_ecs_total_RF(
+        test_RF_data,
+        valid_tcr_ecs_diagnosis_results["tcr_yr"],
+        valid_tcr_ecs_diagnosis_results["ecs_yr"],
+    )
+    test_time = test_RF_data.index.values
+    for year_to_break in [
+        test_time[0],
+        test_time[15],
+        test_time[115],
+        test_time[-1] - 100,
+        test_time[-1],
+    ]:
+        broken_CO2_data = test_RF_data.copy()
+        broken_CO2_data.loc[year_to_break] = (
+            test_RF_data.loc[year_to_break] * 1.01 + 0.01
+        )
+        with pytest.raises(
+            ValueError, match=r"The TCR/ECS total radiative forcing looks wrong.*"
+        ):
+            magicc_base._check_tcr_ecs_total_RF(
+                broken_CO2_data,
+                valid_tcr_ecs_diagnosis_results["tcr_yr"],
+                valid_tcr_ecs_diagnosis_results["ecs_yr"],
+            )
+
+
+def test_check_tcr_ecs_temp(valid_tcr_ecs_diagnosis_results, magicc_base):
+    test_temp_data = valid_tcr_ecs_diagnosis_results["mock_results"]["SURFACE_TEMP"][
+        "GLOBAL"
+    ]
+    magicc_base._check_tcr_ecs_temp(test_temp_data)
+
+    test_time = test_temp_data.index.values
+    for year_to_break in [
+        test_time[3],
+        test_time[15],
+        test_time[115],
+        test_time[-1] - 100,
+        test_time[-1],
+    ]:
+        broken_temp_data = test_temp_data.copy()
+        broken_temp_data.loc[year_to_break] = (
+            test_temp_data.loc[year_to_break - 1] - 0.1
+        )
+        with pytest.raises(
+            ValueError,
+            match=r"The TCR/ECS surface temperature looks wrong, it decreases",
+        ):
+            magicc_base._check_tcr_ecs_temp(broken_temp_data)
+
+
+# integration test (i.e. actually runs magicc) hence slow
+@pytest.mark.slow
+def test_integration_diagnose_tcr_ecs(package):
+    actual_result = package.diagnose_tcr_ecs()
+    assert isinstance(actual_result, dict)
+    assert "tcr" in actual_result
+    assert "ecs" in actual_result
+    assert actual_result["tcr"] < actual_result["ecs"]
+    if isinstance(package, MAGICC6):
+        assert (
+            actual_result["tcr"] == 1.9733976000000002
```

Comment:
That's probably too accurate ... :-)
Maybe there should be an "almost_equal" comparison or so be used?

Reply:
for a stable MAGICC version (i.e. when this is always done with pymagicc's version of MAGICC6), that should never change right? Hence the accuracy is ok/a useful backup to make sure we didn't swap the binary somewhere?

Reply:
Not sure, I think the "...0002" is just coming from Python reading the data and this might be annoying to debug in CI if it changed randomly.

Reply:
smart